src/core/constraint_template.jl

```diff
@@ -87,7 +87,17 @@ end
 
 ""
 function constraint_tp_theta_ref(pm::GenericPowerModel, i::Int; nw::Int=pm.cnw, ph::Int=pm.cph)
-    constraint_tp_theta_ref(pm, nw, ph, i)
+    bus = ref(pm, nw, :bus, i)
+    varef = bus["va"][ph]
+    constraint_tp_theta_ref(pm, nw, ph, i, varef)
```

Comment:
To match the semantics of the refrence bus in transmission systems, this constraint should fix the phase angles to standardized values that only depend on phase/conductor configuration.  If you want to fix the angles to specific values in the data, then the constraint should have a different name.  However, I am curious in what use case such a constraint is useful.

Reply:
For a three-phase case an explicit neutral, a voltage reference vector abcn may look like [Va, Vb, Vc, 0] (similar to (6) in [1]). So four conductors but phases 120 degrees shifted. In split-phase the angles may be 90/180 degrees. I'm not sure what to do to keep compatibility, except by predefining a set of possible configurations (e.g. 3-phase no neutral, 3-phase explicit neutral, split-phase, etc.) and how they map to a reference bus definition?  

[1] Ciric, R. M., Feltrin, A. P., & Ochoa, L. F. (2003). Power flow in four-wire distribution networks - general approach. IEEE Trans. Power Syst., 18(4), 1283–1290.

Reply:
Fixing the values from the datafile is a resonable solution while you are exploring the formulations.  In the long term me feeling is that we should add some more metadata to the network data, which indicates different conductor configurations.  This info can be used by the refrence bus constraint to set the variables correctly.

Reply:
Ok, good approach. So which of the two do you want now?
`@constraint(pm.model, va == wraptopi(2 * pi / nphases * (1-h)))`
or 
`@constraint(pm.model, va == varef)` with varef set by the parser?

Reply:
My recommendation would be to,
1) `constraint_tp_theta_ref` to follow the angle offset standard you think is best.  I believe that is, `@constraint(pm.model, va == wraptopi(2 * pi / nphases * (1-h)))`
2) add a new constraint (e.g. `constraint_mc_theta_setpoint`) for multi-conductor theta setpoint, which uses `@constraint(pm.model, va == varef)` as specified in the data file.
3) Add a new problem specification (e.g. `mc_opf` or `tpn_opf`), which supports an explicit neutral and uses `constraint_mc_theta_setpoint` instead of `constraint_tp_theta_ref`